store/sqlstore/user_store.go

```diff
@@ -232,17 +232,15 @@ func (us SqlUserStore) ResetLastPictureUpdate(userId string) *model.AppError {
 	return nil
 }
 
-func (us SqlUserStore) UpdateUpdateAt(userId string) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		curTime := model.GetMillis()
-
-		if _, err := us.GetMaster().Exec("UPDATE Users SET UpdateAt = :Time WHERE Id = :UserId", map[string]interface{}{"Time": curTime, "UserId": userId}); err != nil {
-			result.Err = model.NewAppError("SqlUserStore.UpdateUpdateAt", "store.sql_user.update_update.app_error", nil, "user_id="+userId, http.StatusInternalServerError)
-			return
-		}
+func (us SqlUserStore) UpdateUpdateAt(userId string) (*model.User, *model.AppError) {
+	curTime := model.GetMillis()
+	var user model.User
 
-		result.Data = curTime
-	})
+	if _, err := us.GetMaster().Exec("UPDATE Users SET UpdateAt = :Time WHERE Id = :UserId", map[string]interface{}{"Time": curTime, "UserId": userId}); err != nil {
+		return nil, model.NewAppError("SqlUserStore.UpdateUpdateAt", "store.sql_user.update_update.app_error", nil, "user_id="+userId, http.StatusInternalServerError)
+	} else {
```

Comment:
Because the if always returns, you don't need this `else` here, you can just write the content of the else at top level of the function (that is the idiomatic way in go).